ecological/autoconfig.py

```diff
@@ -145,16 +154,80 @@ def get(self, wanted_type: WantedType) -> Union[WantedType, Any]:
         return value
 
 
-class ConfigMeta(type):
+class Autoload(enum.Enum):
+    CLASS = "CLASS"
+    OBJECT = "OBJECT"
+    NEVER = "NEVER"
+
+
+@dataclasses.dataclass
+class Options:
+    prefix: Optional[str] = None
+    autoload: Autoload = Autoload.CLASS
+
+    @classmethod
+    def from_metaclass_kwargs(cls, metaclass_kwargs: Dict) -> "Options":
+        options_kwargs = {}
+        for field in dataclasses.fields(cls):
+            value = metaclass_kwargs.pop(field.name, None)
+            if value is None:
+                continue
+
+            options_kwargs[field.name] = value
+
+        try:
+            return cls(**options_kwargs)
+        except TypeError as e:
+            raise ValueError(
+                f"Invalid options for Config class: {metaclass_kwargs}."
+            ) from e
+
+
+class Config:
     """
-    Metaclass that does the "magic" behind ``AutoConfig``.
+    When ``Config`` sub classes are created ``Ecological`` will automatically set it's
```

Comment:
`it's` → `their`

I know it was there before but now it's in your diff :stuck_out_tongue_winking_eye: